pkg/autodiscovery/configresolver/configresolver.go

```diff
@@ -448,11 +448,11 @@ func tagsAdder(tags []string) func(interface{}) error {
 		}
 
 		if typedTree, ok := tree.(map[interface{}]interface{}); ok {
-			tagList, _ := typedTree["tags"].([]string)
+			tagList, _ := typedTree["tags"].([]interface{})
 			// Use a set to remove duplicates
 			tagSet := make(map[string]struct{})
 			for _, t := range tagList {
-				tagSet[t] = struct{}{}
+				tagSet[t.(string)] = struct{}{}
```

Comment:
same, I think we should make sure the casting was successful

Reply:
I added a log.